pkg/dogstatsd/server.go

```diff
@@ -341,6 +347,39 @@ func NewServer(demultiplexer aggregator.Demultiplexer, serverless bool) (*Server
 		}
 	}
 
+	// Creating a new logger
+	cfg := seelogCfg.NewSeelogConfig(
+		"dogstatsd",
+		"info",
+		"common",
+		"",
+		"%Date %Time | [DOGSTATSD_STATS] | [%LEV] %Msg ",
+		false)
+
+	// Configuring the log file path
+	logFile := config.Datadog.GetString("dogstatsd_log_file")
+	if logFile == "" {
+		logFile = constants.DefaultDogstatsDLogFile
+	}
+
+	// Configuring max roll for log file, if dogstatsd_log_file_max_rolls env var is not set within datadog.yaml then default value is 3
+	dogstatsd_log_file_max_rolls := config.Datadog.GetInt64("dogstatsd_log_file_max_rolls")
+	if dogstatsd_log_file_max_rolls == 0 {
+		config.Datadog.Set("dogstatsd_log_file_max_rolls", dogstatsd_log_file_max_rolls)
+	}
+
+	// Configure the logger
+	cfg.EnableFileLogging(logFile, config.Datadog.GetSizeInBytes("log_file_max_size"), uint(config.Datadog.GetInt("dogstatsd_log_file_max_rolls")))
+
+	seelogConfigStr, err := cfg.Render()
+	if err != nil {
+		log.Error(err)
```

Comment:
What happens if configuration fails to render? Is it still safe to pass to `LoggerFromConfigAsString`?

Reply:
if the config fails then the default behavior would be logging to console but it won't save to any log file so i guess minimal performance impact will be the result

Reply:
+ I also added a null check when the the log is supposed to to be used in case the error cause a panic error.